src/config_parser.h

```diff
@@ -69,6 +69,7 @@ class Parameter_t
   HBTReal TreeAllocFactor;
   HBTReal TreeNodeOpenAngle;
   HBTInt TreeMinNumOfCells;
+  HBTInt ParticleGroupNullId; // SWIFT reads this as an int, so we cannot overflow
```

Comment:
Are SWIFT FOF IDs always ints?

Reply:
This is for the value used to indicate hostless particles. This value is currently assumed to be an int, at least when reading the SWIFT reader. @MatthieuSchaller, any comments about this?

Reply:
SWIFT group ids are size_t.

Reply:
@jchelly following from this comment, the FOF group IDs in swift are always `long long`. We are thus thinking of making the value of `GroupNullId` default to `2^63 - 1`, rather than its current value of `2^31-1`. Should we then change all HostId related values to `long long` instead of `HBTInt`?

Reply:
I think the quick fix here would be to check the value of NullGroupId (loaded as a long long regardless of what HBTInt is) and abort if it doesn't fit in a HBTInt.

I wouldn't want to change the type of halo IDs throughout HBT. This issue is specific to swift so we should deal with it in the swift reader.

We could modify the swift reader to always read halo IDs as long long, replace instances of the null ID in the arrays with -1 as used in the rest of HBT and then convert to HBTInt and continue if all of the values fit. Then we could run with 4 byte HBTInts even if NullGroupId=2^63-1. But I'm not sure we care about having that capability so maybe it's not worth the work.

Reply:
As long as 2^31-1 is not a problem for flamingo-10k don't do extra work.

Reply:
I added a check for overflow in NullGroupId based on the type of HBTInt, as John suggested. I think this is sufficient for now.

Reply:
Where abouts is the check?